### DISCUSSION TITLE
Trouble with device type assignment

### DISCUSSION THREAD

**Discussion by BriacD** [netbox-community/netbox#14528] [Q&A]
The interfaces that are part of my Cisco WS-3560-CX-12PC-S are not applying to my device that I have assigned that type to.
I'm confused if I'm doing something wrong here or if I'm running in to a bug. Can't say I've seen this before.
Any tips appreciated.

Device type configured with:
![configured_device_type_interfaces](https://github.com/netbox-community/netbox/assets/10373814/d87bbb85-1a21-48fe-968e-00a1a75a6942)

A switch with the the device type applied:
![device_type_assigned](https://github.com/netbox-community/netbox/assets/10373814/0b09a76b-7172-4f07-a81c-4ef35c92b7c1)

The interfaces I see on that device:
![device_type_assigned_interfaces](https://github.com/netbox-community/netbox/assets/10373814/1d6fe060-1019-48d3-a65f-821c630c5700)

**Comment by abhi1693** [comment]
Did you change the interfaces on the device type after creating the device?

**Reply by BriacD** [reply]
Yes, I did.

**Reply by abhi1693** [reply]
Once the device is instantiated, the changes from the device type doesn't affect the device

**Reply by candlerb** [reply]
The device type components are copied to the device when the device is first instantiated.

There is a script which can copy missing interfaces (i.e. an interface with that name exists in the template but not the device):
https://github.com/netbox-community/customizations/blob/master/scripts/add_device_type_components.py

**Reply by jeremystretch** [reply]
You can easily rename these interfaces by using the "bulk rename" function.

**Reply by BriacD** [reply]
Thanks, I'll give this a shot.

**Reply by BriacD** [reply]
Yeah my only issue is I have over 300 devices I need to change. Is there a way to programmatically achieve this via a bulk rename? Is it an accessible function via the API or python module?

**Reply by candlerb** [reply]
If you've not done any configuration so far on these devices, it may be easier just to delete the interfaces and then repopulate them with the script linked above (or even just delete the devices and recreate them entirely).
